src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+            arguments[0]?.Expression as LiteralExpressionSyntax;
+
+        protected override InterpolatedStringExpressionSyntax GetInterpolatedString(string text) =>
+            (InterpolatedStringExpressionSyntax)ParseExpression("$" + text);
+
+        protected override string GetText(SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            var stringToken = ((LiteralExpressionSyntax)arguments[0].Expression).Token;
+            var text = stringToken.ToString();
+
+            if (stringToken.IsVerbatimStringLiteral())
+            {
+                // We need to escape braces as this is an ambiguous case in interpolated strings that is 
+                // not ambiguous in verbatim strings
+                text = text.Replace(@"'{'", @"'{{'").Replace(@"'}'", @"'}}'");
+            }
+
+            return text;
+        }
+
+        protected override bool IsArgumentListCorrect(
+            InvocationExpressionSyntax invocation,
+            ISymbol invocationSymbol,
+            ImmutableArray<ISymbol> formatMethods,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            if (invocation.ArgumentList != null &&
+                invocation.ArgumentList.Arguments.Count >= 2 &&
+                invocation.ArgumentList.Arguments[0].Expression.IsKind(SyntaxKind.StringLiteralExpression))
+            {
+                // We do not want to substitute the expression if it is being passed to params array argument
+                // Example: 
+                // string[] args;
+                // String.Format("{0}{1}{2}", args);
+                return IsArgumentListNotPassingArrayToParams(
+                    invocation.ArgumentList.Arguments[1].Expression,
+                    invocationSymbol,
+                    formatMethods,
+                    semanticModel,
+                    cancellationToken);
+            }
+
+            return false;
+        }
+
+        protected override bool IsStringLiteral(LiteralExpressionSyntax firstArgument) =>
+            firstArgument?.Token.IsKind(SyntaxKind.StringLiteralToken) == true;
+
+        protected override InterpolatedStringExpressionSyntax VisitArguments(
+            ImmutableArray<ExpressionSyntax> expandedArguments,
+            InterpolatedStringExpressionSyntax interpolatedString)
+        {
+            return interpolatedString.ReplaceNodes(interpolatedString.Contents, (oldNode, newNode) =>
+            {
+                var node = newNode as InterpolationSyntax;
+                if (node == null)
+                {
+                    return newNode;
+                }
+
+                var literalExpression = node.Expression as LiteralExpressionSyntax;
+                if (literalExpression != null && literalExpression.IsKind(SyntaxKind.NumericLiteralExpression))
+                {
+                    var index = (int)literalExpression.Token.Value;
```

Comment:
this may not work if the numeric literal is something like 0L.   Verify it's actually an int and add test for this.